bootstrap/controllers/ck8sconfig_controller.go

```diff
@@ -258,7 +258,13 @@ func (r *CK8sConfigReconciler) joinControlplane(ctx context.Context, scope *Scop
 		return err
 	}
 
-	snapInstallData := r.resolveInPlaceUpgradeRelease(machine)
+	snapInstallData := r.setSnapInstallDataFromSpec(scope.Config.Spec)
+
+	inPlaceInstallData := r.resolveInPlaceUpgradeRelease(machine)
+
+	if inPlaceInstallData != (cloudinit.SnapInstallData{}) {
```

Comment:
Right, resolveInPlaceUpgrade returns an "empty" struct. I'll change it to direct field comparison so that devs. don't need to know the default initialization value of SnapInstallData fields.